src/common/dcsvg.cpp

```diff
@@ -108,6 +119,157 @@ wxString wxBrushString(wxColour c, int style = wxBRUSHSTYLE_SOLID)
     return s;
 }
 
+wxString wxGetPenPattern(wxPen& pen)
+{
+    wxString s;
+
+    // The length of the dashes and gaps have a constant factor.
+    // Dots have a width of 2, short dashes 10, long dashes 15 and gaps 8 (5 for dots).
+    // When the pen width increases, lines become thicker and unrecognizable.
+    // Multiplying with 1/3th of the width creates line styles matching the appearance of wxDC.
+    // The pen width is not used to modify user provided dash styles.
+    float w = pen.GetWidth();
+    if (pen.GetWidth() == 0)
+        w = 1;
+    w = w / 3;
+
+    switch (pen.GetStyle())
+    {
+        case wxPENSTYLE_DOT:
+            s.Printf(wxT("stroke-dasharray=\"%f,%f\" "), w * 2, w * 5);
```

Comment:
In wxString string literals (at least in these newly introduced) it would be better to use wxS macro.